src/PointwiseFunctions/Xcts/CenterOfMass.hpp

```diff
@@ -17,33 +17,66 @@ namespace Xcts {
  * \cite Ossokine2015yla):
  *
  * \begin{equation}
- *   C_{CoM}^i
- *   = \frac{1}{M_{ADM}} \int_{S_\infty} \frac{3}{8\pi} \psi^4 n^i \, d\bar{A}.
+ *   C_\text{CoM}^i = \frac{3}{8 \pi M_\text{ADM}}
+ *               \oint_{S_\infty} \psi^4 n^i n^j \, dS_j
  * \end{equation}
  *
- * Note that we don't include the ADM mass $M_{ADM}$ in this integrand. After
+ * \note We don't include the ADM mass $M_{ADM}$ in this integrand. After
  * integrating the result of this function, you have to divide by $M_{ADM}$.
- * See `Xcts::adm_mass_surface_integrand` for details on how to calculate
- * $M_{ADM}$.
+ *
+ * \see `Xcts::adm_mass_surface_integrand`
  *
  * \warning This integral assumes that the conformal metric falls off to
  * flatness faster than $1/r^2$. That means that it cannot be directly used
  * with the Kerr-Schild metric, which falls off as $1/r$. This is not a problem
  * for XCTS with Superposed Kerr-Schild (SKS) because of the exponential
  * fall-off terms.
  *
- * \param result output buffer for the surface integrand
+ * \param result output pointer
  * \param conformal_factor the conformal factor $\psi$
  * \param unit_normal the outward-pointing unit normal $n^i = x^i / r$
  */
 void center_of_mass_surface_integrand(
+    gsl::not_null<tnsr::II<DataVector, 3>*> result,
+    const Scalar<DataVector>& conformal_factor,
+    const tnsr::I<DataVector, 3>& unit_normal);
+
+/// Return-by-value overload
+tnsr::II<DataVector, 3> center_of_mass_surface_integrand(
+    const Scalar<DataVector>& conformal_factor,
+    const tnsr::I<DataVector, 3>& unit_normal);
+/// @}
+
+/// @{
+/*!
+ * \brief Volume integrand for the center of mass calculation.
+ *
+ * We cast the center of mass as an infinite volume integral by applying Gauss'
+ * law on the surface integral defined in `center_of_mass_surface_integrand`:
+ *
+ * \begin{equation}
+ *   C_\text{CoM}^i = \frac{3}{8 \pi M_\text{ADM}}
+ *               \int_{V_\infty} 4 \psi^3 \partial_j \psi n^i n^j \, dV.
+ * \end{equation}
+ *
+ * \see `center_of_mass_surface_integrand`
+ *
+ * \param result output pointer
+ * \param conformal_factor the conformal factor $\psi$
+ * \param deriv_conformal_factor the gradient of the conformal factor
+ * $\partial_i \psi$
+ * \param unit_normal the outward-pointing unit normal $n^i = x^i / r$
+ */
+void center_of_mass_volume_integrand(
     gsl::not_null<tnsr::I<DataVector, 3>*> result,
     const Scalar<DataVector>& conformal_factor,
+    const tnsr::i<DataVector, 3, Frame::Inertial>& deriv_conformal_factor,
     const tnsr::I<DataVector, 3>& unit_normal);
```

Comment:
I suggest you pass the coordinates $x^i$ and the Euclidean radius $r$ to this function and to the surface integrand. That seems easiest, because then you don't have to compute the Euclidean normal outside this function and you can include the correct powers of $r$ within this function.

Reply:
Since $r$ is just `magnitude(coords)`, do you think we could just pass the coordinates $x^i$? Should we do the same with the surface integrands?

Reply:
You can also just pass in the coords, but then you may have to repeat the magnitude computation in multiple places. Not a big deal, but can be avoided.